src/flash-source-buffer.js

```diff
@@ -77,18 +79,32 @@ export default class FlashSourceBuffer extends videojs.EventTarget {
     this.updating = false;
     this.timestampOffset_ = 0;
 
-    // TS to FLV transmuxer
-    this.segmentParser_ = new flv.Transmuxer();
-    this.segmentParser_.on('data', this.receiveBuffer_.bind(this));
     encodedHeader = window.btoa(
       String.fromCharCode.apply(
         null,
         Array.prototype.slice.call(
-          this.segmentParser_.getFlvHeader()
+          flv.getFlvHeader()
         )
       )
     );
-    this.mediaSource_.swfObj.vjs_appendBuffer(encodedHeader);
+
+    /* eslint-disable camelcase */
+    window.vjs_flashEncodedHeader_ = function() {
+      delete window.vjs_flashEncodedHeader_;
+      return encodedHeader;
+    };
+    /* eslint-enable camelcase */
+
+    this.mediaSource_.swfObj.vjs_appendChunkReady('vjs_flashEncodedHeader_');
+
+    // TS to FLV transmuxer
+    this.transmuxer_ = work(transmuxWorker);
```

Comment:
No it isn't. I see that for the html media source, the worker is terminated on the `sourceclose` event. Does flash emit this event as well and would it be the right time to terminate the worker for flash as well?